src/BizHawk.Emulation.Cores/Waterbox/NymaCore.cs

```diff
@@ -181,7 +182,7 @@ public abstract unsafe partial class NymaCore : WaterboxCore
 					// if (deterministic)
 					// 	throw new InvalidOperationException("Internal error: Core set a frame thread proc in deterministic mode");
 					Console.WriteLine($"Setting up waterbox thread for {_frameThreadPtr}");
-					_frameThreadStart = CallingConventionAdapters.Waterbox.GetDelegateForFunctionPointer<Action>(_frameThreadPtr);
+					_frameThreadStart = CallingConventionAdapters.WaterboxDepartureOnly.GetDelegateForFunctionPointer<Action>(_frameThreadPtr);

```

Comment:
This is actually very sneaky.  The way `_frameThreadPtr` was passed back, it doesn't get wrapped by `thunks.rs` and so it runs on a host stack and any attempt for it to call one of the registered callbacks, or make a syscall, will crash.  This is fine because it's just a consequence of our fake not-at-all-real "threading" support.  Maybe add a comment...